### PR TITLE
New: Added support for notify popup graphic (fixes #16)

### LINKED ISSUE #16: Add support for notify popup graphic
This was an item raised in #13 which I think needs further input so I'm raising this as a separate issue. Looking at the existing graphic support in notify popups is handled differently in Adapt.

**Core notify** renders the graphic within the [`.notify__section`](https://github.com/adaptlearning/adapt-contrib-core/blob/2c455d7b5cf3c52c4457378d3c6c30b800b2d8c1/templates/notifyPopup.hbs#L30) with the positioning dependant on the `_imageAlignment` (whether to render before or after `.notify__body`). See screenshot below.

tutor-popup

**Hotgraphic** and **Hotgrid** don't render the graphic within `.notify__section`, instead this is just an empty element. The difference being these components use their own popup templates. In addition, the positioning `_imageAlignment` is handled via [CSS flex styles](https://github.com/adaptlearning/adapt-contrib-hotgraphic/blob/63a02f77bf6b9e87a754fad8cd4d1649250a131a/less/hotgraphicPopup.less#L87) instead of setting the DOM order. See screenshot below.

hotgraphic-popup

Worth noting, although the implementation is handled differently, the graphic config is consistent e.g. question `_feedback`, Hotgraphic and Hotgrid `_items`.

```
      "_imageAlignment": "right",
      "_graphic": {
          "src": "course/en/images/example.jpg",
          "alt": "",
          "attribution": "",
          "_classes": ""
      }
```

Thoughts on how we should go about this? 

An ideal for me, we wouldn't have popup templates in plugins or at least not for the properties shared across notify popups. Although this is a bigger issue than Hint as it would be good to uniform this across the plugins. 

In regard to `_imageAlignment`, I'd be in favour of setting the DOM order instead of using CSS flex styles as this is better a11y.

### PR DESCRIPTION
**Pull request by kirsty-hames**
Hint popup uses Core Notify template instead of plugin sub view to utilise `_imageAlignment` and `_graphic` properties.

**Test PR**

Apply the following config to any component within your course _component.json_. Note, you'll need to update the image `_src` to an existing asset:
<pre>
    "_hint": {
        "_isEnabled": true,
        "title": "Adapt Hint",
        "altTitle": "Alt Hint title text",
        "body": "An extension that adds a small, clickable icon to a component that displays additional information.",
        "_imageAlignment": "right",
        "_graphic": {
            "_src": "course/en/images/example.jpg",
            "alt": "",
            "attribution": ""
        },
        "_button": {
            "_iconClass": "icon-question",
            "_alignIconRight": true,
            "text": "Hint",
            "ariaLabel": ""
        }
    }
</pre>
Select the Hint button within your course. Your chosen image should display with the layout set by `_imageAlignment`.

Fixes https://github.com/cgkineo/adapt-hint/issues/16

This PR requires Core v6.20.5 https://github.com/adaptlearning/adapt-contrib-core/releases/tag/v6.20.5

### CODE REVIEW

**Review by oliverfoster — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -30,6 +30,21 @@ The attributes listed below are used in *components.json* to configure **Hint**,
 
 >**body** (string): The body text displayed in the popup.
 
+>**\_imageAlignment** (string):
+Defines the alignment of the popup image. Left: Image aligned to the left of the text area. Top: Image aligned above the text area. Right: Image aligned to the right of the text area. Bottom: Image aligned below the text area. The default alignment is `right`.
+
+>**\_graphic** (object):
+The image displayed in the popup. It contains the following settings:
+
+>>**_src** (string):
```
Comment (guywillis) [inline README.md]: Missing a `\`, >>**\_src** (string):

File: properties.schema [hunk 2]
```diff
@@ -80,6 +80,48 @@
...
+                  "help": "Defines the alignment of the pop up image. Left: Image aligned to the left of the text area. Top: Image aligned above the text area. Right: Image aligned to the right of the text area. Bottom: Image aligned below the text area. The default alignment is `right`."
+                },
+                "_graphic": {
+                  "type": "object",
+                  "required": true,
+                  "title": "Graphic",
+                  "properties": {
+                    "_src": {
+                      "type": "string",
+                      "required": true,
```
Comment (guywillis) [inline properties.schema]: I would suggest we make the `_graphic` object `_src` required property false so we can have full width text only as an option

File: schema/component.schema.json [hunk 3]
```diff
@@ -46,6 +46,57 @@
...
+                "bottom"
+              ],
+              "_backboneForms": "Select"
+            },
+            "_graphic": {
+              "type": "object",
+              "title": "Graphic",
+              "default": {},
+              "required": [
+                "_src"
```
Comment (guywillis) [inline schema/component.schema.json]: Same comment as before but for new schema

**Review by guywillis — COMMENTED** [review]
Some comments left for feedback

File: README.md (hunk 1 above)
Comment (kirsty-hames) [reply README.md]: Amended in [cfff4d2](https://github.com/cgkineo/adapt-hint/commit/cfff4d21f8b79e92742a8ce78d363dd02a57cff0)

File: properties.schema (hunk 2 above)
Comment (kirsty-hames) [reply properties.schema]: Amended in 483c09b0d9f332cec391c1972ace7275c20e08b5

File: schema/component.schema.json (hunk 3 above)
Comment (kirsty-hames) [reply schema/component.schema.json]: Amended in 483c09b0d9f332cec391c1972ace7275c20e08b5

**Comment by kirsty-hames** [comment]
> Some comments left for feedback

Thanks @guywillis. Comments have been addressed.

File: properties.schema [hunk 4]
```diff
@@ -80,6 +80,48 @@
...
+                  "type": "string",
+                  "required": false,
+                  "default": "right",
+                  "inputType": {"type":"Select", "options":["left","top","right","bottom"]},
+                  "title": "Image alignment",
+                  "help": "Defines the alignment of the pop up image. Left: Image aligned to the left of the text area. Top: Image aligned above the text area. Right: Image aligned to the right of the text area. Bottom: Image aligned below the text area. The default alignment is `right`."
+                },
+                "_graphic": {
+                  "type": "object",
+                  "required": true,
```
Comment (guywillis) [inline properties.schema]: Not sure if this property is also now not required as we have set the `_src` property to not required?

File: properties.schema [hunk 5]
```diff
@@ -80,6 +80,48 @@
...
+                  "title": "Graphic",
+                  "properties": {
+                    "_src": {
+                      "type": "string",
+                      "required": false,
+                      "default": "",
+                      "inputType": "Asset:image",
+                      "validators": ["required"],
```
Comment (guywillis) [inline properties.schema]: This validator can also be removed

**Review by guywillis — COMMENTED** [review]
Couple more comments

**Comment by kirsty-hames** [comment]
> Couple more comments

Thanks @guywillis. Amended in [8216474](https://github.com/cgkineo/adapt-hint/commit/821647438b8beb4fcf2fa58312d733ba5a103f3c)

**Review by guywillis — APPROVED** [review]
👍  Lovely, thank you

**Pull request merged**